cc-cli/Program.cs

```diff
@@ -21,41 +21,43 @@ class Program
     {
         private static ConsoleColor _defaultConsoleColor = Console.ForegroundColor;
 
-        private static IUpdateService _updater;
         private static ILoggingService _logger;
-        private static IAnalyticsService _analyzer;
-
-        private static LocalStorage _localStorageGlobalSettings;
 
         // Global Settings Constants
+        private const string _storageKeyBase = "cc-cli:global.";
         private const string CHECKFORUPDATES = "check-for-updates";
         private static bool _checkForUpdates = true;
 
         static void Main(string[] args)
         {
             Console.CancelKeyPress += delegate {
-                // call methods to clean up
-                Console.ForegroundColor = _defaultConsoleColor;
+                ExitProgram(false);
             };
 
             ArgumentHandler argHandler = new ArgumentHandler(args);
-            
+
             ServiceProvider provider = ApplicationServiceProvider
                 .CreateServiceProvider(argHandler.ArgData.Debug);
 
             _logger = provider.GetRequiredService<ILoggingService>();
-            _analyzer = provider.GetRequiredService<IAnalyticsService>();
-            _updater = provider.GetRequiredService<IUpdateService>();
+            IUpdateService updater = provider.GetRequiredService<IUpdateService>();
+
+            if(string.IsNullOrEmpty(provider.GetRequiredService<IConfiguration>()["ClientID"]))
+            {
+                ExitProgram(false, $"The \"ClientID\" is invalid or missing, terminating application.");
```

Comment:
Should we add a link to the client registration form perhaps? What is the user supposed to do if they don't know the client ID?

Reply:
@Jjohns-Hypertherm So the clientID is added during development. There is nothing a user can do about it if it is missing.
They would need to get a different version or report the issue to the developer. In out case this should never be possible since the Actions pipeline handles adding in the secrets, but if we ever see the error in testing we will have a clear understanding of what to fix.

Reply:
Thanks for clarifying that. If the user could ever potentially see this message (which it sounds like they won't?), it would be more useful for them if we could advise to "update to a newer version to fix the issue."

Reply:
I will add something to inform the user they need to get a different version.
How does this sound:
$"The \"ClientID\" is invalid or missing, please try a different version."

Reply:
Looks good. I think it should be two sentences though. How about:
$"The "ClientID" is invalid or missing. Please try a different version."